ckanext/search/filters.py

```diff
@@ -61,365 +61,400 @@ def __repr__(self) -> str:
         )
 
 
-def _validate_field_operator(field_name, operator, value, search_schema):
+class FiltersParser:
 
-    if field_name not in search_schema["fields"].keys():
-        return [f"Unknown field: {field_name}"]
+    search_schema: dict = {}
+    filters: Optional[FilterOp] = None
+    errors: Optional[list[str]] = None
+    total_ops: int = 0
 
-    # TODO: check value depending on operation and field type
+    def __init__(
+        self,
+        input_value: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]],
+        search_schema: dict,
+    ) -> None:
 
+        self.search_schema = search_schema
 
-def _is_dict_or_list_of_dicts(value: Any) -> bool:
-    return isinstance(value, dict) or _is_list_of_dicts(value)
+        self.filters, self.errors = self._parse_query_filters(
```

Comment:
You have a class now so your `_parse_query_filters` and other methods could return only filters and when errors are encountered they would be added directly to `self.errors` instead.